src/OrchardCore.Modules/OrchardCore.Media/Indexing/IMediaFileTextProvider.cs

```diff
@@ -0,0 +1,22 @@
+using System.IO;
+
+namespace OrchardCore.Media.Indexing
+{
+    public interface IMediaFileTextProvider
+    {
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="path"></param>
+        /// <returns></returns>
+        /// <remarks>
+        /// <para>
+        /// A separate method is necessary since an empty or null return value from <see cref="GetText(string, Stream)"/>
+        /// can simply mean the file's textual content is empty.
+        /// </para>
+        /// </remarks>
+        bool CanHandle(string path);
+
+        string GetText(string path, Stream fileStream);
```

Comment:
If we have a `path` we can get the stream instead of passing it

Reply:
Not necessarily. Path here can be any identifier of the file, including a Blob Storage URL even. This is only needed to be able to determine whether it's a file type we can handle, by looking at the extension (this is not foolproof by any means but the quickest and easiest way to do it). Hence we also require a `Stream` to get the content of the file, whatever form it's stored.

Reply:
I kind of agree with @hishamco on that one but it's not a big deal.

```C#
    string extension = Path.GetExtension(myFileStream.Name);
```